### PR TITLE
SAA-2239: Default to on-wing if activity also has location id

### PR DESCRIPTION
**Pull request by munslowl**
## Intro

3 changes:

- If migrating an activity which has a location id and a location code containing "WOW" then do not set the location id but do set on-wing to true
- When viewing movement list locations do not included activities that are both on-wing and have a location id
- When viewing movement list items remove any that are both on-wing and have a location id.

:warning: **There are ~500 activities which are on-wing and have a location id - there are no plans to fix the data**

### Movement locations before where C Wing Exercise Yard is both on wing and has a location:
image

### Movement list items where Woodworking is both on wing and has a location:
image

### Movement locations before where C Wing Exercise Yard (removed) is both on wing and has a location:
image

### Movement list items where Woodworking (removed) is both on wing and has a location:
image

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/PrisonerScheduledActivityRepository.kt [hunk 1]
```diff
@@ -69,6 +69,7 @@ interface PrisonerScheduledActivityRepository : JpaRepository<PrisonerScheduledA
     SELECT sa FROM PrisonerScheduledActivity sa 
     WHERE (sa.prisonCode = :prisonCode
     AND sa.sessionDate = :date
+    AND sa.onWing = false
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/PrisonerScheduledActivityRepository.kt:72]: Do not show any activities in movement list where `onWing` is true

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/MigrateActivityService.kt [hunk 2]
```diff
@@ -223,7 +223,7 @@ class MigrateActivityService(
     }.apply {
       addSchedule(
         description = this.summary,
-        internalLocation = request.internalLocationId?.let { prisonApiClient.getLocation(it).block() },
+        internalLocation = if (!onWing) request.internalLocationId?.let { prisonApiClient.getLocation(it).block() } else null,
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/MigrateActivityService.kt:226]: Do not set location id if on-wing

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/InternalLocationService.kt [hunk 3]
```diff
@@ -109,6 +109,7 @@ class InternalLocationService(
 
   private fun getLocationActivitiesMap(prisonCode: String, date: LocalDate, timeSlot: TimeSlot?): Map<Long, PrisonerScheduledActivity> =
     prisonerScheduledActivityRepository.findByPrisonCodeAndDateAndTimeSlot(prisonCode, date, timeSlot)
+      .filterNot { it.onWing }
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/InternalLocationService.kt:112]: Do not show any locations in movement list if on wing

**Review by dafhoustonmoj — APPROVED** [review]
LGTM

**Pull request merged**